api/v1alpha3/conditions_consts.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+const (
+	MachineConditionCount      = 2
+	ControlPlaneConditionCount = 3
+	ClusterConditionCount      = 2
+)
+
+// AzureCluster Conditions and Reasons
+const (
+	NetworkInfrastructureReadyCondition = "NetworkInfrastructureReady"
+	APIServerProvisioningReason         = "APIServerProvisioning"
```

Comment:
Having `APIServerProvisioning` as a step for `NetworkInfrastructureReady` confuses me a little bit...

Is this meant to provision a load balancer endpoint for the API server? Should we rename this "LoadBalancerProvisioning" or something more specific? FYI, this is the current proposal for VSphereCluster https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/c66649e1d9c8ec2597aee8fb81f0a61ca604d433/api/v1alpha3/condition_consts.go#L24-L33

Reply:
This was the naming used buy the code so I was trying to be consistent, will check the MS documentation

Reply:
I think LoadBalancerProvisioning makes more sense here since this is referring to the infrastructure status

Reply:
thank you Cecile

Reply:
renamed to LoadBalancerProvisioning